### PR TITLE
[Zam/Michael] Fix NPE when AsciiSequenceView::toString

### PR DESCRIPTION
**Pull request by michaelszymczak**
Fixed NPE when AsciiSequenceView::toString and charAt inconsistencies.

* Followed the Liskov Substitution Principle

* AsciiSequenceView implements CharSequence and by its contract it should not throw NPE under any circumstances when toString called

* charAt should respect the boundaries

### CODE REVIEW

**Comment by RichardWarburton** [comment]
Thanks.

File: artio-codecs/src/main/java/uk/co/real_logic/artio/util/AsciiSequenceView.java [hunk 1]
```diff
@@ -19,6 +19,10 @@ public int length()
 
     public char charAt(final int index)
     {
+        if (buffer == null || index < 0 || index >= length)
+        {
+            throw new IndexOutOfBoundsException("AsciiSequenceView index out of range: " + index);
+        }
```
[the remark this answers is not in the document]
Comment (mjpt777) [reply artio-codecs/src/main/java/uk/co/real_logic/artio/util/AsciiSequenceView.java]: It should also be StringIndexOutOfBoundsException.

**Comment by mjpt777** [comment]
I've reflected the updates in Agrona in preparation for the move. https://github.com/real-logic/agrona/blob/master/agrona/src/main/java/org/agrona/AsciiSequenceView.java

**Comment by michaelszymczak** [comment]
Agree. While awaiting the move, I've mirrored the Agrona's implementation so that we know that the replacement will be seamless. See #195 .

**Pull request merged**